src/QsCompiler/Transformations/Monomorphization.cs

```diff
@@ -332,9 +348,10 @@ public static QsCallable Apply(QsCallable current, GetConcreteIdentifierFunc get
 
             public class TransformationState
             {
-                public readonly Concretization CurrentParamTypes = new Concretization();
+                public readonly Stack<TypeParameterResolutions> CurrentTypeParamResolutions = new Stack<TypeParameterResolutions>();
```

Comment:
It is not completely clear to me why there was a change from `Dictionary` to `Stack` here, can you please expand a bit more on the reason for this change?

Reply:
Individual type parameter resolutions are dictionaries, and in exploring an expression the transformation can come across many of these dictionaries. These dictionaries need to be combined to create a more complete picture of how the type parameters are resolved in that expression. Previously this file was trying to do that combining itself as it went along, only holding keeping track of the current calculated dictionary. However there is a utility that performs this combining now, which is used by the call graph. Since we are calculating these resolutions in order to compare them against the resolutions found by the call graph, it is safer if these resolutions were calculated the same way, using the same tool, which is the TypeParameterCombination utility.

Reply:
In short, it was changed to a stack to match how the call graph handles type parameter resolutions.

Reply:
Makes sense, thanks for the explanation Scott!